### DISCUSSION TITLE
Pls add account

### DISCUSSION THREAD

**Discussion by PietroMB** [PaulWoitaschek/Voice#1319] [Ideas]
App-Version:
Phone Name: Samsung Tab S4 & Xiami MI MIX 2S (LineasgeOS)
Android Version: 9

Pls add the account in app to share position between multiple devices.
PS great work this project is awsome!

**Comment by change-agent** [comment]
Ditto – I would love to have book position synced to help with switching between devices or restoring to a new device

**Comment by jtagcat** [comment]
As I mentioned in #939 (duplicate), instead of having an account and building a cloud for Voice, dealign with personal data laws (and in some countries, get a licence), you could just have it be in a file, sync/backup/restore that yourself.

**Comment by TheStaticTurtle** [comment]
There could be a custom server that someone could deploy to something like heroku or docker. That way Voice wouldn't be holding the data but only the server owner

**Comment by jtagcat** [comment]
> There could be a custom server that someone could deploy to something like heroku or docker. That way Voice wouldn't be holding the data but only the server owner

There could, but the thing is there's 10x more work than what can be done. Yes you could add WebDAV support for Nextcloud or build your own protocol with your own server. If you want it done, you'll have to do it yourself or pay up. The 'have a syncable dir or file' solves all problems, but with many tens of times less work. There is no need for the deployment buzzwords, when fine solutions exist, plus custom server would require more setup.
Only thing you can't really have? The Spotify real-time switch output devices. For that you may just as well sell your data and add audiobooks as tracks to Spotify.

**Comment by TheStaticTurtle** [comment]
You seem very angry, I was just posting a possible solution I don't need this feature I just really like the app and would like to see it grow

**Comment by HowardWolosky** [comment]
What if this had an opt-in permission for [Google Drive access](https://developers.google.com/drive/android/appfolder), and then stored a sync file in the Google Drive App Folder?  No new account is needed (it's just the user's Google account), and there's no server to maintain (since it's using the user's own Google Drive storage space).  Access would be opt-in (users who don't want to provide Voice access to their Google Drive can simply deny the permission when its requested).

The feature isn't completely straightforward, because it would be a new sync file format (it would have to be able to identify the same file despite the filepath since the filename/filepath might differ per-device (possibly using the filehash instead as the unique identifier).

This is an interesting feature and one that I could see myself using as well since we're using multiple devices around the house now to listen to audiobooks during the pandemic.  Happy to start working through more of the design if this is something that you're actually interested in, @PaulWoitaschek....

**Comment by HowardWolosky** [comment]
Isn't this a dupe of #295?

**Comment by jtagcat** [comment]
Almost, but not quite. It's position vs whole library + settings + position.
